File: Problems/Easy/66 . plus-one.py
class Solution(object):
    def plusOne(self, digits):
        if digits == []:  #just for case: digits = [9]
            return [1]
        if digits[-1] == 9:
            digits = self.plusOne(digits[:-1])
            digits.extend([0])
        else:
            digits[-1] += 1
        return digits

File: Problems/Easy/66 . plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    l=digits.length;
    ++digits[l-1];
    while(--l>=0)
    {
       
        if(digits[l] <= 9  ) break;
        else {
            digits[l]=0
            l==0? digits = [1].concat(digits) : digits[l-1]++
            
        }
    }  
    return digits
};
